### DISCUSSION TITLE
Does Vercel provide the ability to lock Deployments? What about setting preview deployment default paths?

### DISCUSSION THREAD

**Discussion by Joroze** [vercel/community#1006] [Help]
Hi!

I had two questions before considering the idea of migrating over to Vercel.

- Does Vercel provide the ability to lock Deployments? 
- Does Vercel provide the ability to set default preview deployment url paths? (for example setting to /login: `https://myproject-i1hgb65v9-company.vercel.app/login`)

My current service provider allows both, which are very useful.

I had trouble finding relevant documentation anywhere.

Thank you!

**Comment by amyegan** [comment]
Hi @Joroze! Can you explain a little more about what you mean? I don't want to give you the wrong information based on faulty assumptions 😅

**Reply by Joroze** [reply]
Hi @amyegan, sure. 

> Does Vercel provide the ability to lock Deployments?

There are edge cases in our app where it'd be easier to have Vercel provide the deployment link with a set path at the end such as "/login" or "/blog/" or "/blog/new". This is something that should be configurable for all previews, or for specific ones by mentioning some command in a PR such as `@vercel preview_path="/login"`

Example: https://docs.netlify.com/site-deploys/deploy-previews/#set-entry-path

> Does Vercel provide the ability to lock Deployments?

I just learned that we can now roll back deployments using not only the CLI, but now the web console too. 

With that said, are we able to lock our production environment to be pointed to a specific deployed build? 

This is something that services such as Netlify allows, which helps in cases where we may have accidentally shipped broken code out, and we can immediately roll back, and additionally, lock production to point to a specific previous deployment while new commits continue to generate new deployments

example: https://docs.netlify.com/site-deploys/manage-deploys/#locked-deploys

**Reply by amyegan** [reply]
Thanks for sharing this! I'm going to need to double check on these, but I'll share what I know so far.

>Does Vercel provide the ability to lock Deployments?

Right now, Enterprise plans have a project administrator role which can prevent those with developer role from updating the production deploy. I know that isn't exactly what you're asking for, but it's the currently available solution. We do have a planned feature that would allow you to do what you're describing, but I don't have an ETA on that.

>Does Vercel provide the ability to set default preview deployment url paths?

I don't know about this one. I have not seen this feature before, but I'm asking the team for more information.
